vm/systemSmartContracts/staking.go

```diff
@@ -198,19 +211,31 @@ func (r *stakingSC) slash(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 
 	var registrationData stakingData
 	data := r.eei.GetStorage(args.Arguments[0].Bytes())
-	err := json.Unmarshal(data, registrationData)
+	if data == nil {
+		return vmcommon.UserError
+	}
+
+	err := json.Unmarshal(data, &registrationData)
 	if err != nil {
 		log.Error("unmarshal error on slash function" + err.Error())
 		return vmcommon.UserError
 	}
 
-	if len(data) == 0 {
-		log.Error("slash error: validator was not registered")
-		return vmcommon.UserError
+	if !registrationData.Staked {
+		return vmcommon.Ok
 	}
 
 	operation := big.NewInt(0).Set(registrationData.StakeValue)
```

Comment:
operation is not a good name for this. call it stakeValue

Reply:
Done.